src/game_api/script/script_impl.cpp

```diff
@@ -642,6 +733,43 @@ ScriptImpl::ScriptImpl(std::string script, std::string file, SoundManager* sound
         return (uint16_t)0;
     };
 
+    /// Clears a callback that is specific to an entity.
+    lua["clear_entity_callback"] = [this](int uid, CallbackId cb_id)
+    {
+        clear_entity_hooks.push_back({uid, cb_id});
+    };
+    /// Returns unique id for the callback to be used in [clear_entity_callback](#clear_entity_callback) or `nil` if uid is not valid.
+    /// `uid` has to be the uid of a `Movable` or else stuff will break.
```

Comment:
would it make sense to make the methods on Movable's then?

Reply:
Makes sense from that perspective, but the script has to keep track of which entities it hooked for cleanup on reload (so the movable doesn't call stale code), so moving the callback to `Movable` complicates things quite a bit